main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+var (
+	tempFileName = "trivy-go-plugin-temp.json"
+)
+
+func main() {
+	trivyCommand := os.Args[1 : len(os.Args)-1]
+	tempFilePath := filepath.Join(os.TempDir(), tempFileName)
+	defer removeFile(tempFilePath)
+
+	cmdArgs := append(trivyCommand, "--format", "json", "--output", tempFilePath)
+	if err := exec.Command("trivy", cmdArgs...).Run(); err != nil {
+		log.Fatalf("failed to build report: %v", err)
+	}
+
+}
+func getScanType() string {
+	return os.Args[1]
+}
```

Comment:
i have an idea.
we can try to parse the json as a basic report and, if the error appears we'll try to parse as a k8s report.

Reply:
there will be no error, just empty values will be written if using encoding/json